libslide/src/parser.rs

```diff
@@ -0,0 +1,63 @@
+use crate::scanner::types::Token;
+
+pub enum Expr {
+    Float(f64),
+    Int(i64),
+    BinOp(Box<BinOp>),
+    // I added un op even though I havent implemented + or -
+    UnaryOp(Box<UnaryOp>),
```

Comment:
You actually don't need `Box` here, you only need to hold it dynamically when the precise type of the expression is not known (i.e. in the left/right of a binary op, left comments below). So this can be
```suggestion
    BinOp(BinOp),
    // I added un op even though I havent implemented + or -
    UnaryOp(UnaryOp),
```